docs/source/advanced/processors.rst

```diff
@@ -0,0 +1,126 @@
+
+Processors explained
+--------------------
+
+A processor is a :any:`callable` function that accepts a list of records and returns a (potentially altered) list of records.
+
+Processors are called in the order in which they are passed, after all inlets finish producing their data. The result of each processor is given to the next one, until finally the resulting records continue the transfer normally.
+
+.. _link-outlet-processors:
+
+Link vs Outlet processors
+--------------------------
+
+Databay supports two types of processors, depending on the scope at which they operate.
+
+* :any:`Link processor <Link>` - applied to all records transferred by that link.
+* :any:`Outlet processor <Outlet>` - applied only to records consumed by the particular outlet.
+
+This distinction can be used to determine at which level a particular processor is to be applied.
+
+.. image:: /_static/images/databay_processors.png
+
+Observe in the diagram above that the *Outlet A* will receive records modified only by the Link Processor. At the same time, the *Outlet B* will receive records modified first by the Link Processor, then by the Outlet Processor.
+
+For example:
+
+* :any:`Link processor <Link>` - A filtering processor that removes duplicate records produced by an inlet could be applied to all records at link level.
+
+.. code-block:: python
+
+    def remove_duplicates(records: List[Record]):
+        result = []
+        for record in records:
+            if record not in result:
+                result.append(record)
+        return result
+
+    link = Link(..., processors=remove_duplicates)
+
+* :any:`Outlet processor <Outlet>` - A filtering processor that filters out records already existing in a CSV file could be applied only to the CsvOutlet, preventing duplicate records from being written to a CSV file, yet otherwise allowing all records to be consumed by the other outlets in the link.
+
+.. code-block:: python
+
+    def filter_existing(records: List[Record]):
+        with open(os.fspath('./data/records.csv'), 'r') as f:
+            reader = csv.DictReader(csv_file)
+            existing = []
+            for row in reader:
+                for key, value in row.items():
+                    existing.append(value)
+
+        result = []
+        for record in records:
+            if record.payload not in existing:
+                result.append(record)
+        return result
+
+    csv_outlet = CsvOutlet(..., processors=filter_existing)
+    link = Link(inlets, csv_outlet, ...)
+
+Link processors are used before :any:`Batchers <batchers>`, while Outlet processors are used after.
+
+
+Best practices
+--------------
+
+.. rubric:: Responsibility
+
+Databay doesn't make any further assumptions about processors - you can implement any type of processors that may suit your needs. This also means Databay will not ensure the records aren't corrupted by the processors, therefore you need to be conscious of what each processor do to the data.
```

Comment:
hey @cdpierse can you check my grammar here?

Is it:

> therefore you need to be conscious of what each processor **do** to the data

or 

> therefore you need to be conscious of what each processor **does** to the data

Just read this and am not sure 😯

Reply:
Missed this q in the review sorry! I think **does** would be most appropriate in this context.

Reply:
thanks buddy 👍 thought so!